### ISSUE TITLE
Add version check to CI

### ISSUE THREAD

**Issue by gavv** [roc-streaming/roc-toolkit#497]
We already have CI check that the code is properly formatted: [`check-formatting.sh`](https://github.com/roc-streaming/roc-toolkit/blob/master/scripts/ci_checks/linux-checks/check-formatting.sh).

It would be nice to add `check-version.sh`, which would verify that version declared in src/public_api/include/roc/version.h is similar as the latest git tag reported by `git describe --tags --abbrev=0`.

This check should be run only when running CI build for tags that match `v*`.

We already have a script to parse version from header, which we can use in this check: [`scripts/scons_helpers/parse-version.py`](https://github.com/roc-streaming/roc-toolkit/blob/master/scripts/scons_helpers/parse-version.py).

Github workflow is here: https://github.com/roc-streaming/roc-toolkit/blob/master/.github/workflows/build.yml

**Comment by dheerajd5** [comment]
Hey, can I work on this issue?

**Comment by gavv** [comment]
You're welcome, thanks!

**Comment by dheerajd5** [comment]
Hey so my code it isn't passing the `android29-x86_64/macos` tests, I noticed that these tests also failed on the last build workflow run on the main repo, so I was thinking maybe it didn't have anything to do with the changes I made.
Other than that, it is passing all the checks. 
I created a new check- `version-check` which does the required functionality.

I had to fetch all the tags manually, so it could affect workflow times a bit, I wasn't able to find a workaround to this, when I try to fetch tags through checkout, it doesn't show up.

Here is my workflow run. [Link to workflow](https://github.com/dheerajd5/roc-toolkit/actions/runs/5905736760)
Also I'm new to contributing, so in the future, do I mention these things here itself or in my PR?
